code/game/atoms_movable.dm

```diff
@@ -20,10 +20,25 @@
 
 	var/auto_init = 1
 
+// This is used in the map loader to defer initialization once all entities
+// are placed, so that pipes and window spawners correctly function
+// when plunked down mid-game
+var/list/dirty_z_levels = list()
 /atom/movable/New()
 	. = ..()
 	areaMaster = get_area_master(src)
-	if(auto_init && ticker && ticker.current_state == GAME_STATE_PLAYING)
+	var/turf/T = get_turf(src)
+
+	// If you're wondering what goofery this is, this is for things that need the environment
+	// around them set up - like `air_update_turf` and the like
+	if(!auto_init)
+		//no-op

```

Comment:
Should I condense this into an impenetrable massive single if conditional that's repeated a whole bunch